File: threedollar-domain/src/main/java/com/depromeet/threedollar/domain/domain/user/UserCreator.java
package com.depromeet.threedollar.domain.domain.user;

import com.depromeet.threedollar.common.docs.ObjectMother;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@ObjectMother
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class UserCreator {

    public static User create(String socialId, UserSocialType socialType, String name) {
        return User.newInstance(socialId, socialType, name);
    }

}

File: threedollar-domain/src/testFixtures/java/com/depromeet/threedollar/domain/user/domain/user/UserCreator.kt
package com.depromeet.threedollar.domain.user.domain.user

import com.depromeet.threedollar.domain.user.domain.TestFixture

@TestFixture
object UserCreator {

    @JvmStatic
    fun create(
        socialId: String,
        socialType: UserSocialType,
        name: String
    ): User {
        return User.builder()
            .socialId(socialId)
            .socialType(socialType)
            .name(name)
            .build()
    }

}
